core-v1/src/main/scala/algebra/Expr.scala

```diff
@@ -296,6 +297,19 @@ object Expr {
       copy(debugging = debugging)
   }
 
+  // TODO: Add infix option and other metadata to a special case class type
+  // TODO: Should there be a separate Transform? expression for extracting a value from justified wrapper classes?
+  //       Or should we just use the Select() expression?
+  //       ... turns out we need to handle the state and not just the value for certain operations
+  final case class CustomFunction[-I, +O : OP, OP[_]](
+    functionName: String,
+    function: I => O,
+    debugging: Debugging[I, Any] = NoDebugging,
+  ) extends Expr[I, O, OP]("customFunction") {
+    override def visit[G[-_, +_]](v: Visitor[G, OP]): G[I, O] = v.visitCustomFunction(this)
+    override def withDebugging(debugging: Debugging[Any, Any]): Expr[I, O, OP] = copy(debugging = debugging)
+  }
```

Comment:
I really don't like this hacky escape hatch, but it has proven to be very useful. I think we'll need something like this for the user of the library, no matter what. I think having good metadata here would alleviate my concerns.

I definitely need to document this class though.